bundles/io/org.eclipse.smarthome.io.net.test/src/test/java/org/eclipse/smarthome/io/net/http/internal/WebClientFactoryTest.java

```diff
@@ -46,57 +55,96 @@
     @Mock
     private TrustManagerProvider trustmanagerProvider;
 
+    @Mock
+    private ExtensibleTrustManager extensibleTrustManager;
+
     @Before
     public void setup() {
         initMocks(this);
         webClientFactory = new WebClientFactoryImpl();
         webClientFactory.setTrustmanagerProvider(trustmanagerProvider);
+        webClientFactory.setExtensibleTrustManager(extensibleTrustManager);
+
+        webClientFactory.activate(createConfigMap(10, 200, 60, 5, 10, 60));
+    }
+
+    @After
+    public void tearDown() {
+        webClientFactory.deactivate();
     }
 
     @Test
     public void testGetClients() throws Exception {
-        webClientFactory.activate(createConfigMap(10, 200, 60, 5, 10, 60));
-
         HttpClient httpClient = webClientFactory.getCommonHttpClient();
         WebSocketClient webSocketClient = webClientFactory.getCommonWebSocketClient();
 
         assertThat(httpClient, is(notNullValue()));
         assertThat(webSocketClient, is(notNullValue()));
+    }
 
-        webClientFactory.deactivate();
+    @Test
+    public void testCommonClientUsesExtensibleTrustManager() throws Exception {
+        ArgumentCaptor<X509Certificate[]> certificateChainCaptor = ArgumentCaptor.forClass(X509Certificate[].class);
+        ArgumentCaptor<SSLEngine> sslEngineCaptor = ArgumentCaptor.forClass(SSLEngine.class);
+        HttpClient httpClient = webClientFactory.getCommonHttpClient();
```

Comment:
The pattern here is that this is the only test involving the commonClient which is actually always executed.

Reply:
And in one of the commits I attempted to mark this test as ignored, I had the same problem locally sometimes. Depending on the firewalling / security settings of the jdk.

Reply:
Even more interesting I did ignore the second one going to TEST_URL but not the first one